mono/metadata/gc.c

```diff
@@ -622,70 +629,73 @@ ves_icall_System_GC_WaitForPendingFinalizers (void)
 }
 
 void
-ves_icall_System_GC_register_ephemeron_array (MonoObject *array)
+ves_icall_System_GC_register_ephemeron_array (MonoObjectHandle array, MonoError *error)
 {
-#ifdef HAVE_SGEN_GC
-	if (!mono_gc_ephemeron_array_add (array)) {
-		mono_set_pending_exception (mono_object_domain (array)->out_of_memory_ex);
-		return;
-	}
-#endif
+	if (!mono_gc_ephemeron_array_add (MONO_HANDLE_RAW (array)))
```

Comment:
Why not just keep the #ifdef here instead of adding a stub to Boehm?

Reply:
ifdef is bad and to be reduced. Variation generally better at link time or runtime (command line switch), in order to compile and/or link fewer times. Unless for extreme perf reasons.